pkg/ocm/config.go

```diff
@@ -39,6 +42,46 @@ var URLAliases = map[string]string{
 	"crc":         "https://clusters-service.apps-crc.testing",
 }
 
+func ValidOCMUrlAliases() []string {
+	keys := make([]string, 0, len(URLAliases))
+	for k := range URLAliases {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+// URL Precedent (from highest priority to lowest priority):
+//  1. runtime `--env` cli arg (key found in `urlAliases`)
+//  2. runtime `--env` cli arg (non-empty string)
+//  3. config file `URL` value (non-empty string)
+//  4. sdk.DefaultURL
+//
+// Finally, it will try to url.ParseRequestURI the resolved URL to make sure it's a valid URL.
+func ResolveGatewayURL(optionalParsedCliFlagValue string, optionalParsedConfig *config.Config) (string, error) {
+	gatewayURL := sdk.DefaultURL
+	source := "default"
+	if optionalParsedCliFlagValue != "" {
+		gatewayURL = optionalParsedCliFlagValue
+		source = "flag"
+		if _, ok := URLAliases[optionalParsedCliFlagValue]; ok {
+			gatewayURL = URLAliases[optionalParsedCliFlagValue]
```

Comment:
i noticed in `GetEnv()` theres a `fedramp.URLAliases`? So i guess we need to support that too?

Reply:
Do we expect this change in FedRAMP as well?
cc: @tylercreller

Reply:
If resolve gateway URL is going to be a main entry point for any authentication paths, we will need to take into account FedRAMP URLs.

If this is specifically only for region discovery flows, it wouldn't apply.

Reply:
I have not heard plans to deploy regionalized services to fedramp, however i do expect it to happen at some point. So whether we need to worry about rosa's fedramp concept or not is more about not introducing regressions. I would look at how `rosa login --govcloud` works today and think through how our changes to env, the new rh-region flag and behavior change of falling back to cfg.URL when none is provided (simplified precise behavior there for brevity) might interact with it. 

Same with the `GetEnv()` function.

Reply:
If FedRAMP is enabled, we use the env variable there (as opposed to gatewyURL here), so this should not affect FedRAMP. For an edge case where env is not provided (since we made it optional now), I've added the previous implementation (i.e. defaulting to prod env) back. 
As for GetEnv(), FedRAMP aliases are different than normal ones, so that would also work as expected.